### DISCUSSION TITLE
Updating the SPI boot data from Linux

### DISCUSSION THREAD

**Discussion by albertbu** [OE4T/meta-tegra#1675] [Q&A]
Hi, 
I'm working on a swupdate integration with our yocto build. Is there a way to burn all the bct* and mb* stuff from Linux? RCM is able to do that, also from linux, so I'd like to know how to reproduce this behavior locally

**Comment by tschoop** [comment]
From within the Linux system you do not have access to directly access QSPI flash (because of security reasons).
 
But the way to do an OTA update of the low level bootloaders (within swupdate in a A/B redundant device) is 
- to create an UEFI capsule (which contains all parts for the low level: MB1, MB2, UEFI, DT overlays) within your yocto build
- then you can integrate the capsule file as an additional swupdate part to get it onto your system and in the right place
- next you can put the manual steps to initiate the UEFI capsule update into your lua postinstall scripting
the rest runs automatically:
- after the final reboot a startup into the current slot will reach UEFI
- UEFI will initiate the capsule update into the QSPI partitions of the spare slot
- and finally UEFI switches the slot to the spare slot and starts over into it, so finally you boot into new bootloaders, new UEFI and your new Linux system

here is some doc:
- build in yocto: integrate tegra-uefi-capsules into your image
- manual steps to initiate UEFI capsule update: https://docs.nvidia.com/jetson/archives/r35.4.1/DeveloperGuide/text/SD/Bootloader/UpdateAndRedundancy.html?highlight=capsule#manually-trigger-the-capsule-update

**Reply by ichergui** [reply]
Thanks @tschoop 
+ the following link if using JetPack 6.0 / L4T R36.3.0 GA release.
https://docs.nvidia.com/jetson/archives/r36.3/DeveloperGuide/SD/Bootloader/UpdateAndRedundancy.html#manually-trigger-the-capsule-update

**Comment by dwalkes** [comment]
Cross linking the existing swupdate integration demo at https://github.com/OE4T/tegra-demo-distro?tab=readme-ov-file#update-image-demo as well.
